sds_data_manager/lambda_code/SDSCode/pointing_frame_handler.py

```diff
@@ -0,0 +1,248 @@
+
+    Parameters
+    ----------
+    et_times : numpy.ndarray
+        Array of times between et_start and et_end.
+
+    Returns
+    -------
+    q_avg : np.array
+        Average quaternion.
+    z_eclip_time : list
+        Z-axis of the ECLIPJ2000 frame. Used for plotting.
+    """
+    body_quats = []
+    z_eclip_time = []
+    aggregate = np.zeros((4, 4))
+
+    for tdb in et_times:
+        # Rotation matrix from IMAP spacecraft frame to ECLIPJ2000.
+        # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.pxform
+        body_rots = spice.pxform("IMAP_SPACECRAFT", "ECLIPJ2000", tdb)
+        # Convert rotation matrix to quaternion.
+        # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.m2q
+        body_quat = spice.m2q(body_rots)
+        body_quats.append(body_quat)
+        # z-axis of the ECLIPJ2000 frame.
+        z_eclip_time.append(body_rots[:, 2])
+
+        # Standardize the quaternion so that they may be compared.
+        if body_quat[0] < 0:
+            body_quat = -body_quat
+
+        # Aggregate quaternions into a single matrix.
+        aggregate += np.outer(body_quat, body_quat)
+
+    # Reference: Claus Gramkow "On Averaging Rotations"
+    # Link: https://link.springer.com/content/pdf/10.1023/A:1011129215388.pdf
+    aggregate /= len(et_times)
+
+    # Compute eigen values and vectors of the matrix A
+    # Eigenvalues tell you how much "influence" each
+    # direction (eigenvector) has.
+    # The largest eigenvalue corresponds to the direction
+    # that has the most influence.
+    # The eigenvector corresponding to the largest
+    # eigenvalue points in the direction that has the most
+    # combined rotation influence.
+    eigvals, eigvecs = np.linalg.eig(aggregate)
+    # q0: The scalar part of the quaternion.
+    # q1, q2, q3: The vector part of the quaternion.
+    q_avg = eigvecs[:, np.argmax(eigvals)]
+
+    return q_avg, z_eclip_time
+
+
+def create_rotation_matrix(et_times):
+    """Create a rotation matrix.
+
+    Parameters
+    ----------
+    et_times : numpy.ndarray
+        Array of times between et_start and et_end.
+
+    Returns
+    -------
+    rotation_matrix : np.array
+        Rotation matrix.
+    z_avg : np.array
+        Inertial z axis. Used for plotting.
+    """
+    # Averaged quaternions.
+    q_avg, _ = average_quaternions(et_times)
+
+    # Converts the averaged quaternion (q_avg) into a rotation matrix
+    # and get inertial z axis.
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.q2m
+    z_avg = spice.q2m(list(q_avg))[:, 2]
+    # y_avg is perpendicular to both z_avg and the standard Z-axis.
+    y_avg = np.cross(z_avg, [0, 0, 1])
+    # x_avg is perpendicular to y_avg and z_avg.
+    x_avg = np.cross(y_avg, z_avg)
+
+    # Construct the rotation matrix from x_avg, y_avg, z_avg
+    rotation_matrix = np.array([x_avg, y_avg, z_avg])
+
+    return rotation_matrix, z_avg
```

Comment:
```suggestion
    return rotation_matrix
```

Reply:
The return of z_avg was used for plotting in the tests.